src/transaction.js

```diff
@@ -625,24 +793,40 @@ Transaction.prototype.hashForZcashSignature = function (inIndex, prevOutScript,
     var hashSequence = this.getSequenceHash(hashType)
     var hashOutputs = this.getOutputsHash(hashType, inIndex)
     var hashJoinSplits = ZERO
+    var hashShieldedSpends = ZERO
+    var hashShieldedOutputs = ZERO
 
     var bufferWriter
-    var baseBufferSize = 4 * 5 + 32 * 4
+    var baseBufferSize = 0
+    baseBufferSize += 4 * 5
+    baseBufferSize += 32 * 4
     if (inIndex !== VALUE_UINT64_MAX) {
       // If this hash is for a transparent input signature (i.e. not for txTo.joinSplitSig), we need extra space
-      bufferWriter = new BufferWriter(baseBufferSize + 4 * 4 + 32 + varSliceSize(prevOutScript))
-    } else {
-      bufferWriter = new BufferWriter(baseBufferSize)
+      baseBufferSize += 4 * 4
```

Comment:
readability: extra space? Please be more precise (see above comment about explaining what each operation means)